core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -172,15 +172,15 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
 
     @Override
```

Comment:
@javanna You're right, the problem remains that (for BoostingQuery) from the rest perspective at least at the moment is required to have a `"postive"/"negative"` section in the json, but they can parse to `null`, so we need the query builder and validate to allow that. I now changed the doXContent to render out an empty json object when inner query is `null`. That would be similar to a user sending a query like `{ "boosting" : { "positive" : { }, "negative" : { }, "negative_boost" : 0.1, "boost" : 1.0 } }` via rest. It will be accepted, but will itself return a `null` query (same behaviour as in the parser before the refactoring).